src/app/state/pool/pool.effects.ts

```diff
@@ -264,21 +265,28 @@ export class PoolEffects {
   getRoles$ = createEffect(() =>
     this.actions$.pipe(
       ofType(PoolActions.getRoles),
+      tap(() => this.loadingService.show()),
       switchMap(() => from(this.iamService.claimsService.getClaimsByRequester({
           did: this.iamService.signerService.did,
-          isAccepted: true
         })).pipe(
-          map((roles) => roles.filter((item) => item.claimType === 'email.roles.verification.apps.energyweb.iam.ewc')),
-          filter(roles => roles.length === 0),
-          map(() => {
-            this.dialog.open(NotEnroledRoleInfoComponent, {
-              width: '400px',
-              maxWidth: '100%',
-              disableClose: true,
-            });
-          })
+          filter((roles) => {
+            const length = roles.filter((item) =>
+              item.claimType === REQUIRED_ROLE_FOR_STAKING && item.registrationTypes.includes(RegistrationTypes.OnChain)).length;
+            if (length === 0) {
+              this.openNotEnrolledRoleDialog();
```

Comment:
Also, a bit pedantic, but I think we've been using `enroled` with one `l` (not `enrolled`) across SB so far
```suggestion
              this.openNotEnroledRoleDialog();
```

Reply:
haha, i will change it. Webstorm all the time is complaining that it should be `enroll` :D